main.py

```diff
@@ -24,17 +25,130 @@ def setup(self):
         self._flow_mods_sent = OrderedDict()
         self._flow_mods_sent_max_size = FLOWS_DICT_MAX_SIZE
 
+        # Storehouse client to save and restore flow data:
+        self.storehouse = StoreHouse(self.controller)
+
+        # Format of stored flow data:
+        # {'flow_persistence': {'dpid_str': {'flow_list': [
+        #                                     {'command': '<add|delete>',
+        #                                      'flow': {flow_dict}}]}}}
+        self.stored_flows = {}
+        self.resent_flows = set()
+
     def execute(self):
         """Run once on NApp 'start' or in a loop.
 
         The execute method is called by the run method of KytosNApp class.
         Users shouldn't call this method directly.
         """
+        self._load_flows()
 
     def shutdown(self):
         """Shutdown routine of the NApp."""
         log.debug("flow-manager stopping")
 
+    @listen_to('kytos/topology.port.created')
+    def resend_stored_flows(self, event):
+        """Resend stored Flows."""
+        dpid = str(event.content['switch'])
+        switch = self.controller.get_switch_by_dpid(dpid)
+        # This can be a problem because this code is running a thread
+        if dpid in self.resent_flows:
+            log.info(f'Flow already resended to Switch {dpid}')
+            return
+        if dpid in self.stored_flows:
+            flow_list = self.stored_flows[dpid]['flow_list']
+            for flow in flow_list:
+                command = flow['command']
+                flows_dict = {"flows": [flow['flow']]}
+                self._install_flows(command, flows_dict, [switch])
+            self.resent_flows.add(dpid)
+            log.info(f'Flows resent to Switch {dpid}')
+
+    # pylint: disable=attribute-defined-outside-init
+    def _load_flows(self):
+        """Load stored flows."""
+        try:
+            data = self.storehouse.get_data()['flow_persistence']
+            if 'id' in data:
+                del data['id']
+            self.stored_flows = data
+
+        except KeyError as error:
+            log.debug(f'There are no flows to load: {error}')
+        else:
+            log.info('Flows loaded.')
+
+    @staticmethod
+    def _generate_match_fields(flow):
+        """Generate flow match fields."""
+        match_fields = {}
+        for field in flow:
+            if 'priority' in field:
+                match_fields['priority'] = flow['priority']
+            if 'cookie' in field:
+                match_fields['cookie'] = flow['cookie']
+            if 'match' in field:
+                match_fields.update(flow['match'])
+        return match_fields
+
+    def _is_equal_flows(self, flow_1, flow_2):
+        """Check if two flows are equal."""
+        flow_1_match_fields = self._generate_match_fields(flow_1)
+        flow_2_match_fields = self._generate_match_fields(flow_2)
+        if flow_1_match_fields == flow_2_match_fields:
+            return True
+        return False
```

Comment:
Suggestions:
* Rename `_is_equal_flows()` to `fields_match()`
* Move both methods to outside the Main class

Another option would be to override the `__eq__` in the `Flow` class, if that's the type of object that you're comparing here. If they're just `dicts`, we could think of casting them to `Flows` before comparing.

Reply:
This method was replaced by `__eq__` in https://github.com/kytos/of_core/pull/102